### PR TITLE
Improve config upgrade warning dialog

### PR DESCRIPTION
**Pull request by camilasan**
- Improve the code.
- Fix the check for the file name.
- Do not always display the warning about the backup of config files of older version.

### CODE REVIEW

File: src/libsync/configfile.cpp [hunk 1]
```diff
@@ -50,7 +50,8 @@
 #define DEFAULT_MAX_LOG_LINES 20000
 
 namespace {
-static constexpr char showMainDialogAsNormalWindowC[] = "showMainDialogAsNormalWindow";
+static constexpr char showMainDialogAsNormalWindowC[] = "";
```
Comment (claucambra) [inline src/libsync/configfile.cpp:53]: nitpick: why this change?

**Review by claucambra — APPROVED** [review]
Seems fine, small comment :)

File: src/libsync/configfile.cpp (hunk 1 above)
Comment (mgallien) [reply src/libsync/configfile.cpp:53]: very good catch
sorry for missing it

**Review by mgallien — CHANGES_REQUESTED** [review]
see comment from @claucambra

**Review by mgallien — APPROVED** [review]
thanks :blue_heart:
